<a href="https://colab.research.google.com/github/Shayankr/Video-Face-Recognition/blob/shayan_commit/Frames_Extraction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Here, I have a video to analyze face-recognition model performance.

## Step-1: Extracting pictures/Images frame by frame from video.

`Note: How to find the specific frame-rate?`

For this, later I will decide, during imporovement of accuracy of the designed model. Then, I will take help of various state-of-the art methods used in object-tracking kind of problems.

So, later I will change frame rate to the best value.

In [1]:
# set a particular frame-rate:
frame_rate = 4 #fps

# frame_interval = default_frame_rate/frame_rate

In [2]:
# importing some necessary libraries
import cv2   # for image processing tasks
import os # for file related tasks

Define a function with parameters (video file path, output_folder and frame_rate) results a collection of frames/images inside output_folder.

Images are named as: frame_1, frame_2, frame_3,........., frame_10, frame_12, frame_13, ........ so on.

In [3]:
# define a function

def extract_frames_from_video(video_path, output_folder, frame_rate):

    # Create output folder if it doesn't exist
    if not os.path.exists(output_folder):
        os.makedirs(output_folder)

    # Open the video file
    video_capture = cv2.VideoCapture(video_path)

    # Get the default frame rate of the video
    default_frame_rate = video_capture.get(cv2.CAP_PROP_FPS)
    # Print the frame rate
    print("Default frame rate:", default_frame_rate)

    frame_interval = default_frame_rate//frame_rate

    success, frame = video_capture.read()
    count = 0
    frame_index = 1

    while success:
        # Extract frames at the specified frame rate
        if count % frame_interval == 0:
            frame_path = os.path.join(output_folder, f"frame_{frame_index}.jpg")
            cv2.imwrite(frame_path, frame)
            frame_index += 1

        success, frame = video_capture.read()
        count += 1

    video_capture.release()
    print(f"Frames are extracted successfully, frame_gap={frame_interval}, fps={frame_rate}")

# Destroy all the windows
# cv2.destroyAllWindows()


Mount G-drive (to get video)

In [4]:
from google.colab import drive
drive.mount('/content/drive')


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [5]:
!pwd #present working directory

/content


In [6]:
dataset_path = "/content/drive/MyDrive/mtp_24/datasets/"

In [7]:
# cd /content/drive/MyDrive/mtp_24/datasets

video_duration = 08min 50sec

In [8]:
# Specify the path to the video file
video_path = "/content/drive/MyDrive/mtp_24/datasets/Video_Om_Shanti_Om.mp4"

# Specify the output folder to save the frames
# output_folder = dataset_path+"output_frames/"
# output_folder = f"{dataset_path}output_frames_{frame_rate}/"
output_folder = dataset_path+"output_frames_"+f"{frame_rate}"


# Specify the frame rate at which to extract frames (e.g., extract 4 frame per second)
frame_rate = 4

# Call the function to extract frames from the video
extract_frames_from_video(video_path, output_folder, frame_rate)


Default frame rate: 25.00198084055554
Frames are extracted successfully, frame_gap=6.0, fps=4


In [9]:
cd "/content/drive/MyDrive/mtp_24/datasets/"

/content/drive/MyDrive/mtp_24/datasets


In [10]:
pwd

'/content/drive/MyDrive/mtp_24/datasets'

In [11]:
ls

output_frames/  output_frames_4/  Video_Om_Shanti_Om.mp4


In [12]:
cd output_frames_4/

/content/drive/MyDrive/mtp_24/datasets/output_frames_4


In [13]:
ls -l | wc -l

2210


In [14]:
len(os.listdir(output_folder))

2209

In [15]:
output_folder

'/content/drive/MyDrive/mtp_24/datasets/output_frames_4'

In [16]:
# os.listdir(output_folder)

At this point of time, I have taken 4 frames per second.

I got total 2210 frames.